linkerd/app/outbound/src/server.rs

```diff
@@ -156,13 +156,12 @@ where
             AllowProfile(config.allow_discovery.clone().into()),
         ))
         .check_new_service::<tcp::Accept, transport::metrics::SensorIo<I>>()
-        .cache(
-            svc::layers().push_on_response(
-                svc::layers()
-                    .push_failfast(dispatch_timeout)
-                    .push_spawn_buffer_with_idle_timeout(buffer_capacity, cache_max_idle_age),
-            ),
+        .push_on_response(
+            svc::layers()
+                .push_failfast(dispatch_timeout)
+                .push_spawn_buffer(buffer_capacity),
         )
+        .push_cache(cache_max_idle_age)
```

Comment:
Looks like the retain layer is missing:
```suggestion
        .push_cache(cache_max_idle_age)
        .push_on_response(http::Retain::layer())
```

Reply:
This isn't an HTTP stack. the TCP server does the retaining needed to keep the TCP service held until the connection is complete.